android/src/main/kotlin/com/mapbox/maps/mapbox_maps/MapInterfaceController.kt

```diff
@@ -162,6 +176,72 @@ class MapInterfaceController(
     }
   }
 
+  @OptIn(MapboxExperimental::class, MapboxDelicateApi::class)
+  override fun queryRenderedFeaturesForTargets(
+    geometry: _RenderedQueryGeometry,
+    targets: List<FeaturesetQueryTarget>,
+    callback: (Result<List<QueriedRenderedFeature>>) -> Unit
+  ) {
+    mapboxMap.queryRenderedFeatures(
+      geometry.toRenderedQueryGeometry(context),
+      targets.map { target ->
+        target.toFeaturesetQueryTarget()
+      }
+    ) {
+      if (it.isError) {
+        callback(Result.failure(Throwable(it.error)))
+      } else {
+        callback(
+          Result.success(
+            it.value!!.map { feature ->
+              feature.toFLTQueriedRenderedFeature()
+            }.toMutableList()
+          )
+        )
+      }
+    }
+  }
+
+  @OptIn(MapboxExperimental::class)
+  override fun queryRenderedFeaturesForFeatureset(
+    geometry: _RenderedQueryGeometry,
+    featureset: FeaturesetDescriptor,
+    filter: String?,
+    callback: (Result<List<FeaturesetFeature>>) -> Unit
+  ) {
+    mapboxMap.queryRenderedFeatures(
+      geometry.toRenderedQueryGeometry(context),
+      featureset.toTypedFeaturesetDescriptor() as TypedFeaturesetDescriptor<*, com.mapbox.maps.interactions.FeaturesetFeature<FeatureState>>,
+      filter?.let { Expression.fromRaw(filter) }
+    ) {
+      callback(
+        Result.success(
+          it.map { feature -> feature.toFLTFeaturesetFeature() }.toMutableList()
+        )
+      )
+    }
+  }
+
+  @OptIn(MapboxExperimental::class)
+  override fun queryRenderedFeaturesInViewport(
```

Comment:
I am not why this is a dedicated function. Moreover implementation should basically be calling `queryRenderedFeaturesForFeatureset` with `RenderedQueryGeometry(ScreenBox(ScreenCoordinate(0.0, 0.0), ScreenCoordinate(mapView.width.toDouble(), mapView.height.toDouble())))` instead of copying this impl

Reply:
I added this as iOS has it (see [here](https://github.com/mapbox/mapbox-maps-ios-internal/blob/bca5e7c559e25d6465a81905f5fad38d1e613bd5/mapbox-maps-ios/Sources/MapboxMaps/Foundation/MapboxMap.swift#L1148)) and do think it is a nice convenience for developers. If we'd like to keep it, I can switch the implementation. 

CC: @persidskiy

Reply:
I think we should remove it from iOS. Implementation seems very straight-forward (use rect 0;0 -> width; height for another similar QRF). At the same time this method does not take into account camera padding which makes it incorrect for Navigation Camera scenario (as they always use padding).

Reply:
This is just a simplified version of qrf, very useful when you want just get something that is on screed but filtered by a filter.

In GLJS, the `geometry` parameter is also optional, and does the same. In iOS it's the different method only because we have a generic Geometry. See https://docs.mapbox.com/mapbox-gl-js/api/map/#map#queryrenderedfeatures

In flutter, ideally we can the geometry optional

Reply:
@persidskiy thanks for more info. WDYT if I'll expose this method on Android similar to JS (for QRF with TypedFeaturesetDescriptor only as it's experimental)? 

<img width="1484" alt="image" src="https://github.com/user-attachments/assets/0ce4901d-bf1c-4d5d-a6d3-a31a7bc6a09c">

Reply:
Sure, that would be perfect 👍

Reply:
Android implementation: https://github.com/mapbox/mapbox-maps-android-internal/pull/2831